### PR TITLE
braintree: Use `Final` for string constants

### PR DESCRIPTION
**Pull request by Avasam**
Based on https://github.com/python/typeshed/pull/11678 to avoid conflicts, so please merge that first

As title says. There's a bunch of classes that braintree uses like string Enums

There's a few strings that were just a bit too long, triggering Y053. although I'm not sure if, given the context they're in, it makes more sense to noqa. Take a look at this commit for comparison: https://github.com/python/typeshed/commit/cea228deb6e0fae042b980f18b4085421e5a69f1

### CODE REVIEW

**Comment by srittau** [comment]
Conflicts now (as expected).

File: stubs/braintree/braintree/apple_pay_card.pyi [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 from braintree.resource import Resource as Resource
 
 class ApplePayCard(Resource):
     class CardType:
-        AmEx: str
-        MasterCard: str
-        Visa: str
+        AmEx: Final = "Apple Pay - American Express"
+        MasterCard: Final = "Apple Pay - MasterCard"
+        Visa: Final = "Apple Pay - Visa"
```
Comment (srittau) [inline stubs/braintree/braintree/apple_pay_card.pyi:9]: Personal opinion: These user facing strings with no further meaning to the computer should just be `Final[str] = ...`, while something like `GatewayRejected: Final = "gateway_rejected"` below makes sense.

**Review by srittau — APPROVED** [review]
LGTM, apart from one personal and completely optional style remark below.

File: stubs/braintree/braintree/apple_pay_card.pyi (hunk 1 above)
Comment (Avasam) [reply stubs/braintree/braintree/apple_pay_card.pyi:9]: Also just a personal opinion: I feel it's easier to indiscriminately copy&paste from source, w/o having to think too much about what arbitrarily counts as an identifier vs a human readable string, which other contributors may not notice and follow in the same way.

There's an argument about the risk of the string changing, causing more churn, but I doubt these are likely to change without a major revision:
- braintree_python won't even remove dead code without a major version bump
- this card type, for example, is part of the result object for a transaction, it's likely users already highly depend on the value of this string for their transaction logs

**Pull request merged**